### ISSUE TITLE
Crash when using isAvailable method before login

### ISSUE THREAD

**Issue by shaoner** [EddyVerbruggen/cordova-plugin-googleplus#95]
Hi,

Using a bare cordova project & the demo/index.html file, I'm able to get a cordova application working, however it crashes when I click "Login with Google+" sometimes.

If I open the application and then click the login button, it works as expected.
However if I press the "isAvailable?" or "Logout" buttons **first**, then pressing "Login with Google+" makes the application crash immediately.

To be clear "window.plugins.googleplus.login()" makes the application crash only if "window.plugins.googleplus.isAvailable()" is called before on android.
I've tested on two different android devices (samsung s4 & samsung tab S), it crashes the same way.

I did the following steps:

#### 2. New cordova project
```shell
# sudo npm update -g cordova
# cordova --version
5.2.0
# cordova create gplustest
# cd gplustest
```

#### 3. Add cordova-plugin-googleplus and android platform
```shell
# cordova plugin add cordova-plugin-googleplus
# cordova platform add android
```

#### 4. Replace the index.html by the one in the demo
```shell
# wget https://raw.githubusercontent.com/EddyVerbruggen/cordova-plugin-googleplus/master/demo/index.html -O www/index.html
``` 

#### 5. Finally build & run the app on a device
```shell
# cordova build
# adb devices
# cordova run --device
```

**Comment by Freundschaft** [comment]
can confirm this behavior

adb logcat gives me this:


E/AndroidRuntime(31351): Process: com.app.sample, PID: 31351
E/AndroidRuntime(31351): java.lang.IllegalArgumentException: the name must not b
e empty: null
E/AndroidRuntime(31351):        at android.accounts.Account.<init>(Account.java:
48)
E/AndroidRuntime(31351):        at com.google.android.gms.auth.GoogleAuthUtil.ge
tToken(Unknown Source)
E/AndroidRuntime(31351):        at nl.xservices.plugins.GooglePlus$2.run(GoogleP
lus.java:195)
E/AndroidRuntime(31351):        at java.util.concurrent.ThreadPoolExecutor.runWo
rker(ThreadPoolExecutor.java:1112)
E/AndroidRuntime(31351):        at java.util.concurrent.ThreadPoolExecutor$Worke
r.run(ThreadPoolExecutor.java:587)
E/AndroidRuntime(31351):        at java.lang.Thread.run(Thread.java:841)
E/AndroidRuntime(  807): FATAL EXCEPTION: pool-1-thread-2
E/AndroidRuntime(  807): Process: com.app.sample, PID: 807
E/AndroidRuntime(  807): java.lang.IllegalArgumentException: the name must not b
e empty: null
E/AndroidRuntime(  807):        at android.accounts.Account.<init>(Account.java:
48)
E/AndroidRuntime(  807):        at com.google.android.gms.auth.GoogleAuthUtil.ge
tToken(Unknown Source)
E/AndroidRuntime(  807):        at nl.xservices.plugins.GooglePlus$2.run(GoogleP
lus.java:195)
E/AndroidRuntime(  807):        at java.util.concurrent.ThreadPoolExecutor.runWo
rker(ThreadPoolExecutor.java:1112)
E/AndroidRuntime(  807):        at java.util.concurrent.ThreadPoolExecutor$Worke
r.run(ThreadPoolExecutor.java:587)
E/AndroidRuntime(  807):        at java.lang.Thread.run(Thread.java:841)

**Comment by zigzag-way** [comment]
Confirm

**Comment by zigzag-way** [comment]
But how can we call any methods if we are not sure that G+ is available?

**Comment by Freundschaft** [comment]
we shouldnt, so its a really annoying bug :(

**Comment by shaoner** [comment]
Hi,

just found the fix in GooglePlus.java, it needs to be cleaned up but basically:
```java
    if (args.optJSONObject(0) != null){
      JSONObject obj = args.getJSONObject(0);
      System.out.println(obj);
      this.webKey = obj.optString(ARGUMENT_WEB_KEY, null);
      this.apiKey = obj.optString(ARGUMENT_ANDROID_KEY, null);
      this.setupScopes(obj.optString(ARGUMENT_SCOPES, null));
    } else { // Fix issue https://github.com/EddyVerbruggen/cordova-plugin-googleplus/issues/95
      this.webKey = null;
      this.apiKey = null;
      this.setupScopes(null);
    }
```

**Comment by shaoner** [comment]
I also noticed that this plugin uses a deprecated dependency as well as deprecated methods (getToken), so it may need to be updated a bit.

**Comment by EddyVerbruggen** [comment]
Confirmed. I'm taking a look (also at @shaoner's fix of course)

**Comment by EddyVerbruggen** [comment]
This has been fixed in 4.0.1 now, sorry for the inconvenience. I missed this while reviewing a recent PR.
